modyn/supervisor/internal/eval/handler.py

```diff
@@ -83,56 +85,88 @@ def get_eval_requests_after_pipeline(self, df_trainings: pd.DataFrame) -> list[E
         assert {"trigger_id", "training_id", "id_model", "first_timestamp", "last_timestamp"} - (
             set(df_trainings.columns)
         ) == set()
+        df_trainings = df_trainings.copy()
+
+        # for sparse datasets we want to use next_training_start-1 as training interval end instead of last_timestamp
+        # as there could be a long gap between the max(sample_time) in one training batch and the min(sample_time) in
+        # the next training batch.
+        # e.g. if we want to train for 1.1.2020-31.12.2020 but only have timestamps on 1.1.2020, last_timestamp
+        # would be 1.1.2020, but the next training would start on 1.1.2021.
+        df_trainings["real_last_timestamp"] = (
+            df_trainings["first_timestamp"].shift(-1, fill_value=df_trainings.iloc[-1]["last_timestamp"] + 1) - 1
+        )
 
         training_intervals: list[tuple[int, int]] = [
-            (row["first_timestamp"], row["last_timestamp"]) for _, row in df_trainings.iterrows()
+            (row["first_timestamp"], row["real_last_timestamp"]) for _, row in df_trainings.iterrows()
         ]
         eval_intervals = self.eval_strategy.get_eval_intervals(training_intervals)
         df_eval_intervals = pd.DataFrame(
             [
-                (eval_interval.start, eval_interval.end, eval_interval.most_recent_model_interval_end_before)
+                (eval_interval.start, eval_interval.end, eval_interval.active_model_trained_before)
                 for eval_interval in eval_intervals
             ],
-            columns=["start", "end", "most_recent_model_interval_end_before"],
+            columns=["start", "end", "active_model_trained_before"],
         )
 
         # now build & potentially reduce the Trainings x EvalIntervals space
         df_cross = df_eval_intervals.merge(df_trainings, how="cross")
 
-        # to check if a combination is the most recent, we first compute if model was trained before the usage starts
-        # last_timestamp (df_trainings) defines the end of the training data;
-        # most_recent_model_interval_end_before (df_eval_intervals): defines center of an eval intervals.
-        df_cross["chronology_ok"] = df_cross["last_timestamp"] <= df_cross["most_recent_model_interval_end_before"]
+        # to check if a combination is the most recent or currently, we first compute if model was trained before
```

Comment:
most recent is told terminology. Is that referring to currently active or trained?